### PR TITLE
helm-v3: Dynamic completion for "helm repo" and "helm plugin"

### PR DESCRIPTION
**Pull request by marckhouzam**
**What this PR does / why we need it**:

This PR adds dynamic shell completion for the commands:
- helm repo remove \<TAB\> (which will select from the names of configured repos)
- helm plugin remove \<TAB\> (which will select from the names of installed plugins)
- helm plugin update \<TAB\> (which will select from the names of installed plugins)

This PR has been tested for backwards compatibility through the acceptance-testing repo.
Tests have been prepared and submitted in PR https://github.com/helm/acceptance-testing/pull/17

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]

**Comment by jdolitsky** [comment]
Can we get a quick review from another maintainer? This change is pretty straightforward

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @marckhouzam

**Pull request merged**
